src/WinClient.hh

```diff
@@ -142,6 +142,8 @@ public:
     unsigned long initial_state, normal_hint_flags, wm_hint_flags;
 
 private:
+    char hostname_char[512];
```

Comment:
this is not needed here on the class-level, its only needed when you actually use system-calls to retrieve the hostname and transfer it into a FbString.

FbString on the other hand _is_ only an alias around std::string:

```
typedef std::string FbString; 
```

so, with that you can use https://en.cppreference.com/w/cpp/string/basic_string/resize before directly using it for the syscall.

but there is no need for that hostname_char[512].